fendermint/testing/materializer/src/concurrency/reporting/tps.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use ethers::prelude::{Block, H256, U256};
+
+pub fn calc_tps(blocks: Vec<Block<H256>>) -> Vec<f64> {
+    let mut tps = Vec::new();
+
+    let offset = 1; // The first block is skipped for lacking a block interval.
+    for i in offset..blocks.len() {
+        let prev_block = &blocks[i - 1];
+        let curr_block = &blocks[i];
+
+        let interval = curr_block.timestamp.saturating_sub(prev_block.timestamp);
+
+        if interval.le(&U256::zero()) {
+            continue;
+        }
```

Comment:
```
 let interval = curr.timestamp.saturating_sub(prev.timestamp);
        if interval.is_zero() {
            continue;
        }
```

this is correct as the subtraction which saturates at zero.

Reply:
Fixed here: https://github.com/consensus-shipyard/ipc/pull/1243/commits/7ef6c36d6f88924533f3ea0bcacef87579225021